### DISCUSSION TITLE
How to open sidenav using expo-router like react-navigaton?

### DISCUSSION THREAD

**Discussion by manojuppala** [expo/router#497] [Q&A]
react-navigation provides DrawerOpen and DrawerClose options to  handle the drawer state. how can I do something similar in expo?

**Comment by wcastand** [comment]
https://expo.github.io/router/docs/features/linking#usenavigation

**Comment by dnishiyama** [comment]
Looks like that link has died. Here is what I ended up doing in "expo-router": "~3.4.10",
```
import { useNavigation } from "expo-router";
import type { DrawerNavigationProp } from "@react-navigation/drawer";
const navigation = useNavigation<DrawerNavigationProp<never>>();
<TouchableOpacity onPress={() => navigation.openDrawer()}>
```

**Reply by dnishiyama** [reply]
And here is the new link: https://docs.expo.dev/router/reference/hooks/#usenavigation
